okapi-core/src/main/java/org/folio/okapi/util/TcpPortWaiting.java

```diff
@@ -34,24 +34,27 @@ public TcpPortWaiting(Vertx vertx, String host, int port) {
   }
 
   private Future<Void> tryConnect(NuProcess process, int count) {
-    NetClientOptions options = new NetClientOptions().setConnectTimeout(MILLISECONDS);
-    NetClient c = vertx.createNetClient(options);
+    // don't use NetClient because container ports are immediately ready, instead check for HTTP response
+    WebClientOptions options = new WebClientOptions().setConnectTimeout(MILLISECONDS);
+    WebClient c = WebClient.create(vertx, options);
     logger.info("tryConnect() host {} port {} count {}", host, port, count);
     Promise<Void> promise = Promise.promise();
-    c.connect(port, host, res -> {
-      if (res.succeeded()) {
-        logger.info("Connected to service at host {} port {} count {}", host, port, count);
-        c.close().onComplete(x -> promise.complete());
-      } else if (count < maxIterations && (process == null || process.isRunning())) {
-        c.close().onComplete(x -> vertx.setTimer((long) (count + 1) * MILLISECONDS,
-            id -> tryConnect(process, count + 1).onComplete(promise)));
-      } else {
-        c.close().onComplete(x ->
+    c.get(port, host, "/").send()
+        .onSuccess(res -> {
+          c.close();
+          logger.info("Connected to service at host {} port {} count {}", host, port, count);
+          promise.complete();
+        })
+        .onFailure(cause -> {
+          c.close();
+          if (count < maxIterations && (process == null || process.isRunning())) {
+            vertx.setTimer((long) (count + 1) * MILLISECONDS,
+                id -> tryConnect(process, count + 1).onComplete(promise));
+          } else {
             promise.fail(messages.getMessage("11501",
-                Integer.toString(port), res.cause().getMessage()))
-        );
-      }
-    });
+                Integer.toString(port), cause.getMessage()));
+          }
+        });
```

Comment:
The WebClient stuff should be moved into a separate method, and Future methods should be used:

```
  private Future<Void> tryConnect(NuProcess process, int count) {
    logger.info("tryConnect() host {} port {} count {}", host, port, count);
    return tryConnect()
        .onSuccess(res -> {
          logger.info("Connected to service at host {} port {} count {}", host, port, count);
        })
        .recover(cause -> {
          if (count < maxIterations && (process == null || process.isRunning())) {
            return Future.future(promise ->
              vertx.setTimer((long) (count + 1) * MILLISECONDS,
                id -> tryConnect(process, count + 1).onComplete(promise)));
          } else {
            return Future.failedFuture(messages.getMessage("11501",
                Integer.toString(port), cause.getMessage()));
          }
        });
  }

  private Future<Void> tryConnect() {
    // don't use NetClient because container ports are immediately ready, instead check for HTTP
    WebClientOptions options = new WebClientOptions().setConnectTimeout(MILLISECONDS);
    WebClient c = WebClient.create(vertx, options);
    return c.get(port, host, "/")
        .send()
        .<Void>mapEmpty()
        .onComplete(result -> c.close());
  }
```

Reply:
More lines, but neat that the WebClient usage is isolated.